### DISCUSSION TITLE
RTMP Connection Success but Streaming not showing on Youtube

### DISCUSSION THREAD

**Discussion by ShubhamBhadauriaImpressico** [HaishinKit/HaishinKit.swift#1361] [Q&A]
The response i am getting see below :
RTMP connection succeeded
Data is nil or not a dictionary
Stream publishing started

**Comment by VE5PA5IAN** [comment]
I am also facing same issue.

2023-27-12 08:03:38.954 [Info] [com.haishinkit.HaishinKit] [RTMPNWSocket.swift:168] stateDidChange(to:) > Connection is ready.
2023-27-12 08:03:38.960 [Info] [com.haishinkit.HaishinKit] [RTMPNWSocket.swift:159] viabilityDidChange(to:) > Connection viability changed to true

nothing on YouTube. 

Tried to run the existing examples, same result. Can you please help? Am I missing anything?

**Reply by shogo4405** [reply]
In most cases, issues like this occur when the RTMP URL and the RTMPStream name are incorrect. Please provide the values you are passing to rtmpConnection.connect and rtmpStream.publish.

**Reply by VE5PA5IAN** [reply]
//rtmpURL initialization - Stream Key is appended to the RTMP URL
@State private var rtmpURL = "rtmp://a.rtmp.youtube.com/live2/<StreamKeyFromYoutTube>"

rtmpConnection.connect(
            rtmpURL
        )
        rtmpStream.publish(
            rtmpURL
        )

**Reply by VE5PA5IAN** [reply]
Below is the line where rtmpStream is published in the LiveViewController of iOS example from the GitHub code -

rtmpStream.publish(Preference.defaultInstance.streamName!)

Preference struct has the entry -

var uri: String? = "rtmp://a.rtmp.youtube.com/live2/<StreamKey>"
    var streamName: String? = "My Test live"

**Reply by shogo4405** [reply]
Pelase set.
```
rtmpConnection.connect("rtmp://a.rtmp.youtube.com/live2")
rtmpStream.publish("your stream name")
```

**Reply by shogo4405** [reply]
`var streamName: String? = "My Test live"`
This stream name is a value specified by YouTube, so setting an arbitrary value won't work.

**Reply by VE5PA5IAN** [reply]
Ok, I was planning to live stream to a specific/ pre-defined stream key. If I need to use a static Stream key, is there a way I can assign it so that my live stream will go live to the specified stream key only?

**Reply by shogo4405** [reply]
Unfortunately, I'm not familiar with the specific details of YouTube's implementation. In the case of YouTube, once a stream key is issued, I understand that it remains fixed unless changed manually.
